src/Serializer.php

```diff
@@ -61,9 +79,20 @@ public function serialize(\Closure $closure)
      */
     public function unserialize($serialized)
     {
-        /** @var SerializableClosure $unserialized */
         $unserialized = unserialize($serialized);
 
+        if ($this->signingKey) {
+            list($data, $signature) = $unserialized;
+            if ($signature !== $this->calculateSignature($data)) {
```

Comment:
You could use: https://packagist.org/packages/realityking/hash_equals

I have given the source a quick look, and it looks good.